Building a Toxicity Classifier using Python (#4881)
section-engineering-education/engineering-education: content/articles/building-a-toxicity-classifier/index.md

### Prerequisites
To follow along with this tutorial, you need to:
- Be familiar with machine learning modeling.
- Use Google Colab or Jupyter Notebook.
- Be familiar with the Jigsaw Multilingual Toxic Comment Classification [dataset](https://www.kaggle.com/c/jigsaw-multilingual-toxic-comment-classification) as the model has been trained on it. 

### Outline
- [The toxicity classifier](#the-toxicity-classifier)
- [Installing the detoxify model and installing the necessary dependencies](#installing-the-detoxify-model-and-installing-the-necessary-dependencies)
- [Performing prediction using the model](#performing-prediction-using-the-model)
- [Deploying the model as an application using Gradio](#deploying-the-model-as-an-application-using-gradio)
- [Wrapping up](wrapping-up)

### The toxicity classifier
Toxicity classier can classify whether a particular text or sentence is toxic or not. Cyberbullying has been an issue ever since the rise of the social media era. Many people have experienced cyberbullying in one way or another. On social media sites, there are a lot of comments that people make, and most of them have a significant amount of negativity and toxicity in them.

> Major technology companies have turned to machine learning to help with the automated classification of toxic content to combat toxic content and online abuse. Indeed, many of these automated systems are deployed today. For example, this has been seen on Twitter, where the company has automatically flagged more than half of all tweets that violate its rules. Just recently, in January 2021, Donald Trump's Twitter account was [permanently suspended](https://blog.twitter.com/en_us/topics/company/2020/suspension) as his account was deemed to cause incitement of violence. A machine learning system aided this.  

Reviewer: ```suggestion
Major technology companies have turned to machine learning to help with the automated classification of toxic content to combat toxic content and online abuse. Indeed, many of these automated systems are deployed today. For example, this has been seen on Twitter, where the company has automatically flagged more than half of all tweets that violate its rules. Just recently, in January 2021, Donald Trump's Twitter account was [permanently suspended](https://blog.twitter.com/en_us/topics/company/2020/suspension) as his account was deemed to cause incitement of violence. This automation is performed by a machine learning system.  
```

To demonstrate how you can use this model, we will use a simple python example to demonstrate how you can implement such a model in your project.

### Installing the detoxify model and installing the necessary dependencies

We begin by performing a quick `pip` install.

```bash
pip install detoxify
```
After installation, we need to import it into our Colab.

```python
from detoxify import Detoxify
```

### Performing prediction using the model
After installing and importing the model, we need to load it into our notebook. We do this by using the `Detoxify` method we imported in the step above. After loading the model, we then save it into a variable known as `predictor`. 

```python
predictor = Detoxify('multilingual')
```

Let's try and perform prediction on a simple sentence to establish whether it's toxic or not.

```python
predictor.predict('Why are you so fat?')
```

> In some modern cultures, calling someone `fat` isn't such a big issue. But in others, people mind find it offensive. Let's see what the model thinks.
> 

Reviewer: ```suggestion
In some modern cultures, calling someone `fat` isn't such a big issue. But in other cultures, they might find it offensive. Let's see what the model thinks.
```
